roles/ceph-osd/tasks/crush_rules.yml

```
@@ -33,15 +33,15 @@
   set_fact:
     info_ceph_default_crush_rule_yaml: "{{ item.stdout | from_json() }}"
   with_items: "{{ info_ceph_default_crush_rule.results }}"
-  when:
-    - inventory_hostname == groups.get(mon_group_name) | last
-    - not item.get('skipped', false)
+  run_once: true
+  when: not item.get('skipped', false)
```

what about 
```suggestion
  when: item.skipped is undefined
```

which is more readable IMO